src/Collector.cs

```diff
@@ -153,5 +143,26 @@ private static Socket Connect()
       _log.Info("Connected to collector.");
       return socket;
     }
+
+    private static bool IsSocketDisconnected (Socket socket)
+    {
+      // Is there any data available?
+      byte[] buffer = null;
+      while (socket.Poll(1, SelectMode.SelectRead))
+        {
+          // If no data is available then socket disconnected
+          if (socket.Available == 0)
+            return true;
+
+          // Clear socket data; we don't care what InstrumentApp sends
+          buffer = buffer ?? new byte[Math.Min(1024, socket.Available)];
+          do
+            {
+              socket.Receive(buffer);
+            }
+          while (socket.Available != 0);
```

Comment:
I think Undead found itself in a position where hello/authenticate were being spammed, and because they are not consuming 'ok', the buffer would fill up with 'ok.  A lot of this seems to exist to prevent that.